SlackBotPrototype/API/SqLitePersistence.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Linq;
+using SQLite;
+
+namespace API
+{
+	// SHORTCUT: Normally we'd break out the repositories into
+	// a file per entity. Keep them together for simplicity here.
+	// 
+	// Also skipping tests for these because we'd need to set up integration tests
+	// with a SQLite database. If we weren't using a embedded file DB like SQLite, testing
+	// against a DB could get complicated.
+	public class SqLitePersistence : IRepository
+	{
+
+		public User Get(string slackId)
+		{
+			if (string.IsNullOrWhiteSpace(slackId))
+			{
+				return null;
+			}
+
+			using (var db = new SQLiteConnection(ConfigConstants.DbName))
+			{
+				return db.Query<User>($"SELECT * FROM User WHERE SlackId = '{slackId}'").FirstOrDefault();
+			}
+		}
+
+		// SHORTCUT: We skip testing this because it's hard to test this in isolation, but it contains
+		// logic for matching up cities which is critical
+		public Location GetLocation(string city, string state)
+		{
+			if (string.IsNullOrWhiteSpace(city) || string.IsNullOrWhiteSpace(state))
+			{
+				return null;
+			}
+
+			using (var db = new SQLiteConnection(ConfigConstants.DbName))
+			{
+				return db.Query<Location>($"SELECT * FROM Location WHERE UPPER(City) = '{city.ToUpper()}' AND UPPER(StateAbbreviation) = '{state.ToUpper()}'").FirstOrDefault();

```

Comment:
If there isn't a record, it'll return `null`. This means we end up having to deal with the the null propagation upstream. Alternatively we could 

* Handle null check here and throw an exception, and upstream catch the exception. 
* Call `First()` instead which will throw an exception if there's no record, and handle exception upstream.

At a high level, I think it's a choice between if we want to check for exceptions upstream or check for nulls.